### PR TITLE
new recipe: libgrapheme

### PR DESCRIPTION
**Pull request by hoanga**
the following recipe adds support for the [libgrapheme](https://libs.suckless.org/libgrapheme/)

checked install on 
```
Haiku shredder 1 hrev56521 Oct  7 2022 07:25:26 x86_64 x86_64 Haiku
```

### CODE REVIEW

File: dev-libs/libgrapheme/libgrapheme-2.0.0.recipe [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+	"
+REQUIRES_devel="
+	libgrapheme$secondaryArchSuffix == $portVersion
+	"
+
+BUILD_REQUIRES="
+	haiku${secondaryArchSuffix}_devel
+	"
+BUILD_PREREQUIRES="
+	cmd:gcc$secondaryArchPrefix
```
Comment (augiedoggie) [inline dev-libs/libgrapheme/libgrapheme-2.0.0.recipe:37]:
```suggestion
	cmd:gcc$secondaryArchSuffix
```

File: dev-libs/libgrapheme/libgrapheme-2.0.0.recipe [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+	haiku$secondaryArchSuffix
+	"
+
+PROVIDES_devel="
+	libgrapheme${secondaryArchSuffix}_devel = $portVersion
+	devel:libgrapheme$secondaryArchSuffix = 4.0.0 compat >= 4
+	"
+REQUIRES_devel="
+	libgrapheme$secondaryArchSuffix == $portVersion
+	"
```
Comment (augiedoggie) [inline dev-libs/libgrapheme/libgrapheme-2.0.0.recipe:31]: You've declared a devel package but there are no commands(`prepareInstalledDevelLib` or `packageEntries`) in the INSTALL section to put the library symlinks, headers,  and man pages there.  So, the devel package ends up being empty.

File: dev-libs/libgrapheme/libgrapheme-2.0.0.recipe [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+SOURCE_FILENAME="libgrapheme-$portVersion.tar.gz"
+CHECKSUM_SHA256="a6e1f7aec86af029dc183e32ab7a8d531fc00e54c44921dbdffc929aceb90dc6"
+PATCHES="libgrapheme-$portVersion.patchset"
+
+ARCHITECTURES="all !x86_gcc2"
+SECONDARY_ARCHITECTURES="x86"
+
+PROVIDES="
+	libgrapheme$secondaryArchSuffix = $portVersion
+	lib:libgrapheme$secondaryArchSuffix = 4.0.0 compat >= 4
```
Comment (augiedoggie) [inline dev-libs/libgrapheme/libgrapheme-2.0.0.recipe:19]: This version doesn't look right.  Where did the 4.0.0 here and down below come from?

**Review by augiedoggie — CHANGES_REQUESTED** [review]

**Comment by Begasus** [comment]
@hoanga could you have a look at the comments?

File: dev-libs/libgrapheme/libgrapheme-2.0.0.recipe (hunk 3 above)
Comment (hoanga) [reply dev-libs/libgrapheme/libgrapheme-2.0.0.recipe]: i think that might have been leftover from a template recipe that i was using as a starting point. i have removed that, thanks for catching that

File: dev-libs/libgrapheme/libgrapheme-2.0.0.recipe (hunk 2 above)
Comment (hoanga) [reply dev-libs/libgrapheme/libgrapheme-2.0.0.recipe]: updated

File: dev-libs/libgrapheme/libgrapheme-2.0.0.recipe (hunk 1 above)
Comment (hoanga) [reply dev-libs/libgrapheme/libgrapheme-2.0.0.recipe]: updated

**Comment by hoanga** [comment]
@Begasus thanks for the reminder on this! somehow the notifications got lost for the update suggestions so didn't realize there was feedback to work on.

File: dev-libs/libgrapheme/patches/libgrapheme-2.0.0.patchset [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
++LDCONFIG = 
+ SH       = sh
+-- 
+2.37.3
+
+
+From b9b0fbe7ef6308a528e70fe4cea049fa8e6cb720 Mon Sep 17 00:00:00 2001
+From: Al Hoang <3811822-hoanga@users.noreply.gitlab.com>
+Date: Sat, 8 Oct 2022 16:03:52 +0000
+Subject: update PREFIX
```
Comment (Begasus) [inline dev-libs/libgrapheme/patches/libgrapheme-2.0.0.patchset:25]: Why update PREFIX? You can use that as you do with the others like `PREFIX=$prefix`

File: dev-libs/libgrapheme/libgrapheme-2.0.0.recipe (hunk 2 above)
Comment (Begasus) [reply dev-libs/libgrapheme/libgrapheme-2.0.0.recipe]: `prepareInstalledDevelLib` still missing ...

File: dev-libs/libgrapheme/libgrapheme-2.0.0.recipe (hunk 3 above)
Comment (Begasus) [reply dev-libs/libgrapheme/libgrapheme-2.0.0.recipe]: libVersion = 2.0.0 compat = 2 maybe use libVersionCompat for this?

File: dev-libs/libgrapheme/patches/libgrapheme-2.0.0.patchset (hunk 4 above)
Comment (Begasus) [reply dev-libs/libgrapheme/patches/libgrapheme-2.0.0.patchset:25]: Checked, second part of the patch isn't needed

File: dev-libs/libgrapheme/libgrapheme-2.0.0.recipe [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+
+INSTALL()
+{
+	make install LIBPREFIX=$libDir INCPREFIX=$includeDir \
+		MANPREFIX=$manDir $jobArgs
+
+	packageEntries devel \
+		$developDir
+
+	rm -rf  $libDir/libgrapheme.a
```
Comment (Begasus) [inline dev-libs/libgrapheme/libgrapheme-2.0.0.recipe:55]: The `.a` file gets moved before `packageEntries devel` if you use `prepareInstalledDevelLib`, so the file is still around, move it before `prepareInstalledDevelLib` (some just keep it in, only the `.la` files are deleted from most recipes)

File: dev-libs/libgrapheme/libgrapheme-2.0.0.recipe (hunk 2 above)
Comment (hoanga) [reply dev-libs/libgrapheme/libgrapheme-2.0.0.recipe]: updated (pardon for leaving that out)

File: dev-libs/libgrapheme/libgrapheme-2.0.0.recipe (hunk 3 above)
Comment (hoanga) [reply dev-libs/libgrapheme/libgrapheme-2.0.0.recipe]: updated

File: dev-libs/libgrapheme/patches/libgrapheme-2.0.0.patchset (hunk 4 above)
Comment (hoanga) [reply dev-libs/libgrapheme/patches/libgrapheme-2.0.0.patchset]: removed patch for PREFIX

File: dev-libs/libgrapheme/libgrapheme-2.0.0.recipe (hunk 5 above)
Comment (hoanga) [reply dev-libs/libgrapheme/libgrapheme-2.0.0.recipe]: updated and removed the .a entry removal

File: dev-libs/libgrapheme/libgrapheme-2.0.0.recipe [hunk 6]
```diff
@@ -0,0 +1,64 @@
+SUMMARY="A unicode string library"
+DESCRIPTION="libgrapheme is an extremely simple freestanding C99 library providing \
+utilities for properly handling strings according to the latest Unicode standard \
+15.0.0. It offers fully Unicode compliant."
+HOMEPAGE="https://libs.suckless.org/libgrapheme"
+COPYRIGHT="2022 Laslo Hunhold"
+LICENSE="ISC"
+REVISION="1"
+SOURCE_URI="https://dl.suckless.org/libgrapheme/libgrapheme-$portVersion.tar.gz"
+SOURCE_FILENAME="libgrapheme-$portVersion.tar.gz"
```
Comment (augiedoggie) [inline dev-libs/libgrapheme/libgrapheme-2.0.0.recipe:10]:
```suggestion
```
Not needed for this recipe

File: dev-libs/libgrapheme/libgrapheme-2.0.0.recipe [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+CHECKSUM_SHA256="a6e1f7aec86af029dc183e32ab7a8d531fc00e54c44921dbdffc929aceb90dc6"
+PATCHES="libgrapheme-$portVersion.patchset"
+
+ARCHITECTURES="all !x86_gcc2"
+SECONDARY_ARCHITECTURES="x86"
+
+libVersion="2.0.0"
```
Comment (augiedoggie) [inline dev-libs/libgrapheme/libgrapheme-2.0.0.recipe:17]:
```suggestion
```
This variable isn't being used

File: dev-libs/libgrapheme/libgrapheme-2.0.0.recipe [hunk 8]
```diff
@@ -0,0 +1,64 @@
...
+
+INSTALL()
+{
+	make install LIBPREFIX=$libDir INCPREFIX=$includeDir \
+		MANPREFIX=$manDir $jobArgs
+
+	prepareInstalledDevelLib libgrapheme
+
+	packageEntries devel \
+		$developDir
```
Comment (augiedoggie) [inline dev-libs/libgrapheme/libgrapheme-2.0.0.recipe:58]:
```suggestion
		$developDir \
		$manDir
```
Man pages aren't needed at runtime and should be moved to the devel package

**Review by augiedoggie — CHANGES_REQUESTED** [review]

**Pull request merged**